update content and examples (#1)
samuelsycamore/material-ui: docs/data/joy/guides/overriding-component-structure/overriding-component-structure.md

## Best practices

> Use `component` or `slotProps.component` prop to override the element by preserving the styles of the slot.

Author: ```suggestion
Use `component` or `slotProps.{slot}.component` prop to override the element by preserving the styles of the slot.
```

> Use `slots` prop to replace the default styled component for that specific slot.

Author: ```suggestion
Use `slots` prop to replace the slot's styles and functionality with your custom component.
```
